### PR TITLE
Clean up fields on page editor

### PR DESCRIPTION
**Pull request by matthinz**
This PR gathers expiration and access control related fields and moves them to the bottom of the Netlify editor page for the "Page" content type:

image

(It also collapses them by default)

I used the `object` widget to do this, which meant that field names in front matter changed slightly (this is why there are so many .md files in this PR):

- `public` is now part of an `access` object
- `expires_at` and `redirect_to` are now part of a `lifecycle` object

I added a `through:` argument to `has_field` to facilitate reading these a little bit. The actual Ruby `Page` model's API is unchanged.

Will close https://trello.com/c/19quaCIz/242-organize-fields-on-the-page-editor-cms-page

### CODE REVIEW

File: app/models/page.rb [hunk 1]
```diff
@@ -117,13 +115,19 @@ def obsolete?
...
+    lifecycle = parsed_file["lifecycle"]
+    return nil if lifecycle.nil?
+
+    if lifecycle["redirect_to"].present?
+      @redirect_page ||= self.class.find_by_path(lifecycle["redirect_to"], base: base, try_index: false).filename
     end
   end
 
   def expired?
-    expires_at.present? && expires_at.past?
```
Comment (rahearn) [inline app/models/page.rb:126]: [suggestion] I think we should revert this change, since we have `expires_at` available through the changes to `has_field` This version just duplicates the `through` logic that exists there.

File: app/models/page.rb [hunk 2]
```diff
@@ -117,13 +115,19 @@ def obsolete?
   end
 
   def redirect_page
-    if redirect_to.present?
-      @redirect_page ||= self.class.find_by_path(redirect_to, base: base, try_index: false).filename
+    lifecycle = parsed_file["lifecycle"]
+    return nil if lifecycle.nil?
+
+    if lifecycle["redirect_to"].present?
+      @redirect_page ||= self.class.find_by_path(lifecycle["redirect_to"], base: base, try_index: false).filename
```
Comment (rahearn) [inline app/models/page.rb:122]: [suggestion] Let's add `redirect_to` to the `has_field` list above and revert this method back to the old version

File: app/views/pages/netlify_config.yaml.erb [hunk 3]
```diff
@@ -132,6 +135,37 @@ collections:
               Order in which this item should be sorted in navigation menus.
               Higher numbers sort after lower numbers.
               If two pages have the same Order, they're sorted alphabetically by title.
+      - label: "Lifecycle"
+        name: "lifecycle"
+        widget: object
+        collapsed: true
+        summary: "Expires: {{fields.expires_at | default('never')}}"
```
Comment (rahearn) [inline app/views/pages/netlify_config.yaml.erb:142]: [question] should the summary also display info about whether "replaced by page" is set?

**Review by rahearn — CHANGES_REQUESTED** [review]
This is way nicer than the previous editor. Just two quick changes to clean up Page and this'll be 👍🏻

File: app/models/page.rb (hunk 1 above)
Comment (matthinz) [reply app/models/page.rb:126]: I will give this a shot

File: app/views/pages/netlify_config.yaml.erb (hunk 3 above)
Comment (matthinz) [reply app/views/pages/netlify_config.yaml.erb:142]: Yes, let me see how that ends up looking (there aren't a ton of formatting controls for the `summary` field)

**Review by rahearn — APPROVED** [review]
🎉
